packages/2d/src/components/Txt.ts

```diff
@@ -105,6 +105,20 @@ export class Txt extends Shape {
     context.restore();
   }
 
+  protected override getCacheBBox(): BBox {
+    const cache = super.getCacheBBox();
+
+    if (!this.element.hasChildNodes) return cache;
+
+    const size = this.computedSize();
+    const range = document.createRange();
+    range.setStartBefore(this.element.firstChild!);
+    range.setEndAfter(this.element.lastChild!);
+    const bbox = range.getBoundingClientRect();
+
+    return new BBox(-size.width / 2, -size.height / 2, bbox.width, bbox.height);
```

Comment:
We should expand the signature of `BBox.fromSizeCentered` and use it here isntead:
```suggestion
    return BBox.fromSizeCentered(bbox);
```

Reply:
Nvm, just noticed it uses the other size in the first few arguments.